pyomo/repn/parameterized_linear.py

```diff
@@ -0,0 +1,400 @@
+            _mult = bool(mult)
+            if not _mult:
+                return
+            if mult == 1:
+                _mult = False
+        except:
+            _mult = True
+
+        const = other.constant
+        try:
+            _const = bool(const)
+        except:
+            _const = True
+
+        if _mult:
+            if _const:
+                self.constant += mult * const
+            if other.linear:
+                _merge_dict(self.linear, mult, other.linear)
+            if other.nonlinear is not None:
+                nl = mult * other.nonlinear
+                if self.nonlinear is None:
+                    self.nonlinear = nl
+                else:
+                    self.nonlinear += nl
+        else:
+            if _const:
+                self.constant += const
+            if other.linear:
+                _merge_dict(self.linear, 1, other.linear)
+            if other.nonlinear is not None:
+                nl = other.nonlinear
+                if self.nonlinear is None:
+                    self.nonlinear = nl
+                else:
+                    self.nonlinear += nl
+
+
+class ParameterizedLinearBeforeChildDispatcher(LinearBeforeChildDispatcher):
+    def __init__(self):
+        super().__init__()
+        self[Var] = self._before_var
+        self[MonomialTermExpression] = self._before_monomial
+        self[LinearExpression] = self._before_linear
+        self[SumExpression] = self._before_general_expression
+
+    @staticmethod
+    def _before_linear(visitor, child):
+        return True, None
+
+    @staticmethod
+    def _before_monomial(visitor, child):
+        return True, None
+
+    @staticmethod
+    def _before_general_expression(visitor, child):
+        return True, None
+
+    @staticmethod
+    def _before_var(visitor, child):
+        _id = id(child)
+        if _id not in visitor.var_map:
+            if child.fixed:
+                return False, (_CONSTANT, visitor.check_constant(child.value, child))
+            if child in visitor.wrt:
+                # pseudo-constant
+                # We aren't treating this Var as a Var for the purposes of this walker
+                return False, (_PSEUDO_CONSTANT, child)
+            # This is a normal situation
+            ParameterizedLinearBeforeChildDispatcher._record_var(visitor, child)
+        ans = visitor.Result()
+        ans.linear[_id] = 1
+        return False, (ExprType.LINEAR, ans)
+
+
+_before_child_dispatcher = ParameterizedLinearBeforeChildDispatcher()
+_exit_node_handlers = copy.deepcopy(linear._exit_node_handlers)
+
+#
+# NEGATION handlers
+#
+
+
+def _handle_negation_pseudo_constant(visitor, node, arg):
+    return (_PSEUDO_CONSTANT, -1 * arg[1])
+
+
+_exit_node_handlers[NegationExpression].update(
+    {(_PSEUDO_CONSTANT,): _handle_negation_pseudo_constant}
+)
+
+
+#
+# PRODUCT handlers
+#
+
+
+def _handle_product_constant_constant(visitor, node, arg1, arg2):
+    return _CONSTANT, arg1[1] * arg2[1]
```

Comment:
Why did you need to override this?

Reply:
Because I didn't include the deprecation path for `0 * nan`. I'll add a comment because it can get un-overridden when that deprecation path goes away.